### PR TITLE
Allow plain merge of data values into a library

### PR DESCRIPTION
**Pull request by pivotaljohn**
- Add keyword argument "plain" to @ytt:library.with_data_values()
- Extract/reuse annotating method from template.DataValuesFile

Addresses #418

### CODE REVIEW

File: pkg/workspace/library_module.go [hunk 1]
```diff
@@ -159,16 +160,33 @@ func (l *libraryValue) WithDataValues(thread *starlark.Thread, f *starlark.Built
...
+		return starlark.None, err
+	}
 
 	dataValues, err := core.NewStarlarkValue(args.Index(0)).AsGoValue()
 	if err != nil {
 		return starlark.None, err
 	}
 
-	valsYAML, err := datavalues.NewEnvelope(&yamlmeta.Document{
+	usePlainMerge, err := core.BoolArg(kwargs, "plain")
```
Comment (cppforlife) [inline pkg/workspace/library_module.go:176]: err check

Comment (pivotaljohn) [reply pkg/workspace/library_module.go:176]: ty.

**Comment by pivotaljohn** [comment]
~~Pulling back to draft because I see an inconsistency:~~

~~To mirror the merge order of the root library, we need to hold separate the _plain_ data values from data value overlays so that the former can be applied after the latter.~~

~~That is, values from `.with_data_values(plain=True)` should be applied _last_ just like values from `--data-values-file` do.~~

_(edit: this is wrong thinking; cleared up below)_

**Comment by pivotaljohn** [comment]
~~So, this gets even more interesting...~~

~~If we ensure that `.with_data_values(plain=True)` has the identical precedence that `--data-values-file` has, then we get weirdness like this:~~

```yaml
#@ lib = library.get("lib")
#@ lib = lib.with_data_values({"from": "plain yaml"}, plain=True)
#@ lib = lib.with_data_values({"from": "overlay"})
--- #@ template.replace(lib.eval())
```

~~would necessarily get you:~~

```yaml
---
from: plain yaml
```

~~Which feels really surprising from the template author's perspective: how is it that the subsequent `.with_data_values()` call is seemingly ignored?!?!~~

...

~~I think we can have both: a mirror of data values precedence between the root and private libraries _and_ keep an intuitive API for template authors...~~
~~if we — instead of adding the `plain` kwarg — introduce a new method: `.set_data_values()`.~~

~~The above would _then_ look like this:~~

```yaml
#@ lib = library.get("lib")
#@ lib = lib.set_data_values({"from": "plain yaml"})
#@ lib = lib.with_data_values({"from": "overlay"})
--- #@ template.replace(lib.eval())
```

_(edit: this is not necessarily; cleared up, below.)_

**Comment by gcheadle-vmware** [comment]
After taking a look through these changes, I like the way you extracted and reused current code.

In response to the most recent comment, I'm having trouble trying to motivate why we need the identical precedence that `--data-values-file` has. When trying to approach these examples from an author's perspective, I would expect that when I programmatically set data values on a library, it would apply in the order specified. Thus I would expect 
```
---
from: overlay
```
for both examples, because as an author, that how I wrote it.

**Comment by pivotaljohn** [comment]
> In response to the most recent comment, I'm having trouble trying to motivate why we need the identical precedence that --data-values-file has. 

Good question. The initial motivation was to try to keep the command-line and programmatic interfaces as similar (if not identical) as possible.

I have just realized a couple of mistakes in my thinking:
- I failed to see `.with_data_values()` as equivalent to all of the command-line flags (i.e. there are many command-line flags because _that_ interface has multiple sources; the programmatic interface really has one: some AST) 💡 
- I misread the order in which the command-line flags were being processed; I got it exactly opposite — I thought the plain DVs were applied last; in fact, they are applied _first_. 🤦🏻‍♂️ 

I'm gonna pull this back into draft and fix it.

File: pkg/cmd/template/data_values_file.go [hunk 2]
```diff
@@ -24,43 +21,10 @@ func NewDataValuesFile(doc *yamlmeta.Document) DataValuesFile {
 func (f DataValuesFile) AsOverlay() (*yamlmeta.Document, error) {
 	doc := f.doc.DeepCopy()
 
-	if yamltemplate.HasTemplating(doc) {
-		return nil, fmt.Errorf("Expected to not find annotations inside data values file " +
-			"(hint: remove comments starting with '#@')")
+	err := overlay.AnnotateForPlainMerge(doc)
+	if err != nil {
+		return nil, fmt.Errorf(err.Error())
```
Comment (cppforlife) [inline pkg/cmd/template/data_values_file.go]: return err

**Comment by cppforlife** [comment]
lgtm

**Comment by pivotaljohn** [comment]
Note, conversations around this work inspired #575. More to come? 😁

**Pull request merged**